t/errlog.t

```diff
@@ -882,3 +882,53 @@ log body:\d{4}/\d{2}/\d{2} \d{2}:\d{2}:\d{2} \[error\] (\d+).*access_by_lua\(ngi
 new line, client: 127.0.0.1, server: localhost, request: "GET /t HTTP/1.1", host: "localhost"
 \z
 --- skip_nginx: 2: <1.11.2
+
+
+
+=== TEST 22: ctx table
+--- http_config
+    lua_capture_error_log 4k;
+--- config
+    location /t {
+        access_by_lua_block {
+            local errlog = require "ngx.errlog"
+            local status, err = errlog.set_filter_level(ngx.WARN)
+            if not status then
+                error(err)
+            end
+
+            ngx.log(ngx.ERR, "-->\n", "new line")
+        }
+
+        content_by_lua_block {
+            local errlog = require "ngx.errlog"
+            local t = {}
+
+            for i = 1, 2 do
+                local res = errlog.get_logs(10, t)
+                ngx.say("maybe log lines: #", #res / 2)
```

Comment:
when we tried to get error logs，reused the table `t`. the method `errlog.get_logs` will set `nil` when there was no error log.

in this case:
first time got data, the `t` value: `t = {4, "xxxx", nil}`.
second time didn't got data, the `t` value: `t= {nil, "xxxx", nil}`, the first `nil` is hole.

if we use `#` to get the table `t` length, it maybe wrong. that's why I wrote `maybe log lines`.